packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -822,6 +838,67 @@ class _DraggableScrollableSheetScrollController extends ScrollController {
   }
 }
 
+class _SheetBallisticScrollActivity extends BallisticScrollActivity {
+  _SheetBallisticScrollActivity(
+    _DraggableScrollableSheetScrollPosition scrollPosition,
+    Simulation simulation,
+  ) : super(scrollPosition, simulation, scrollPosition.context.vsync, false);
+
+  @override
+  _DraggableScrollableSheetScrollPosition get delegate =>
+      super.delegate as _DraggableScrollableSheetScrollPosition;
+
+  /// The relative velocity of the inner content during the ballistic scrolling
+  /// of the outer sheet is always effectively 0. This allows the scroll physics
+  /// to maintain the inner extent within its bounds while the sheet size
+  /// changes.
+  @override
+  double get velocity => 0.0;
+
+  // The velocity of the outer sheet. This is used to transfer velocity when the
+  // widget is updated while a sheet ballistic scrolling activity is underway.
+  double get outerVelocity => super.velocity;
+
+  @override
+  void resetActivity() {
+    // This actually happens before the delegate is updated, so we have to
+    // do our reset in a microtask.
+    //
+    // This may have been a regression introduced long ago.
```

Comment:
I'll probably open a separate issue for this, and spin off a separate PR if advised.

Reply:
Going ahead with https://github.com/flutter/flutter/pull/120053/commits/e07aa68fd2f4ca01086f50017dbed425ddd0d0b1 to fix as part of this PR.